### PR TITLE
3.2.4 Release

### PR DESCRIPTION
**Pull request by cheeseng**
- Further improved support for Scala 3.
- Bumped up to Scala 3.0.0-RC1.

### CODE REVIEW

File: jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala [hunk 1]
```diff
@@ -472,7 +472,10 @@ trait Futures extends PatienceConfiguration {
      * @throws TestFailedException if the future is cancelled, expires, or is still not ready after
      *     the specified timeout has been exceeded
      */
-    def futureValue(implicit config: PatienceConfig, pos: source.Position): T = {
+    // SKIP-DOTTY-START 
+    def futureValue()(implicit config: PatienceConfig, pos: source.Position): T = {
```
Comment (bvenners) [inline jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala:476]: Why are empty parens added here? This is a breaking change to existing source code.

**Comment by catap** [comment]
@cheeseng any chance to include scala3 for scala.js?

File: jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala (hunk 1 above)
Comment (cheeseng) [reply jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala:476]: @bvenners Ah, sorry that was a mistake, the intention was to keep the Scala 2 version as it is, and add it only for Scala 3.

Comment (bvenners) [reply jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala:476]: Hmm, is it needed for Scala 3?

Comment (cheeseng) [reply jvm/core/src/main/scala/org/scalatest/concurrent/Futures.scala:476]: @bvenners Yes, if I remember correctly, that was needed to get our JavaFuturesSpec to build and green, in the following commit: 

https://github.com/scalatest/scalatest/pull/1966/commits/e32243bf9dd4dcdfd59bae7a29ccac209358b30b

The JavaFuturesSpec refused to compile, if I remember correctly it was confused about the implicit parameters.

Having said so, it was 3.0.0-M3, we can try again 3.0.0-RC1 I suppose.

**Comment by cheeseng** [comment]
> @cheeseng any chance to include scala3 for scala.js?

The last time I tried there's an assertion error thrown from the scala-js compiler, so it does not seem like a straight forward thing.  We may try again with 3.0.0-RC1 though.

**Comment by catap** [comment]
as far as I can see you're releasing support of 3.0.0-RC1 and Scala-Native 0.4.0 right now.

I guess it is worth to try to enable support Scala.js at Scala3.

**Pull request merged**